### DISCUSSION TITLE
Troubleshooting to use the hard-verify plugin using commandline in dynamic smart contract verifying with complex constructor arguments inline

### DISCUSSION THREAD

**Discussion by super0605** [NomicFoundation/hardhat#4654] [Troubleshooting]
Regarding the verify contract using hardhat-verify plugin, I am facing some problems to pass the smart contract constructor arguments with command line.
Currently, I am trying to pass the empty array value as the constructor argument in the command line, but they don't accept the array value type.
I am not sure what is the correct format to pass the empty array type.
This is my command line to verify the smart contract.
By the way, the slugList  and priceList  are empty array with [] .
```
const verifyCommand = `npx hardhat verify --network ${blockchainNetwork} ${createReceipt.contractAddress} '${name}' '${symbol}' '${royaltyFee}' '${royaltyRecipient}' '${base_api_url}' '${collectionSize}' '${tokenPrice * 100}' '${slugList}' '${priceList}'`;
This is an example.
npx hardhat verify --network mumbai 0xF9b285e0614b0364F93F5043E1d1FcA9C94ec5F2 'Hardhat Test001' 'hardhat-test001' '1000' '0x3029Cf3FC7c62293700f5851d33eF8A867f95dbE' 'https://api.ecoverse.io/metadata/collections/test-separate-functions/' '100' '0' '[]' '[]'
```
But they don't allow to use '[]'  empty array as the argument.
This is just the problem.
As another solution, I leave the last two constructor arguments with completely empty like the following.
```
npx hardhat verify --network mumbai 0xF9b285e0614b0364F93F5043E1d1FcA9C94ec5F2 'Hardhat Test001' 'hardhat-test001' '1000' '0x3029Cf3FC7c62293700f5851d33eF8A867f95dbE' '[https://api.ecoverse.io/metadata/collections/test-separate-functions/' '100' '0' '' '](https://api.ecoverse.io/metadata/collections/test-separate-functions/'%20'100'%20'0'%20''%20')'
```
But it's also not working with the required array data type in the smart contract constructor validation.

Please help me to solve passing the array arguments inline in the commandline.

**Comment by fvictorio** [comment]
Have you tried [this](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify#complex-arguments)?

**Reply by super0605** [reply]
Hello

This is just error what I am getting when I passed the empty string as the empty[] argument in Hardhat verify command.


```
{"level":"error","message":"Command failed: npx hardhat verify --network mumbai 0x5462AB3A7c0B434741345b8a64e7B017701C9e57 'BrandWallet Test001' 'brandwallet-test001' '1000' '0x3029Cf3FC7c62293700f5851d33eF8A867f95dbE' '[https://api.ecoverse.io/metadata/collections/test-separate-functions/' '100' '0' '](https://api.ecoverse.io/metadata/collections/test-separate-functions/'%20'100'%20'0'%20')' ''\nError in plugin @nomicfoundation/hardhat-verify: Value  cannot be encoded for the parameter slugs.\nEncoder error reason: expected array value\nFor more info run Hardhat with --show-stack-traces\n"}


```
This is my smart contract’s constructor.
```

constructor(
    string memory name,
    string memory symbol,
    uint256 _royaltyFee,
    address _royaltyRecipient,
    string memory baseTokenUri,
    uint256 _collectionSize,
    uint256 _tokenPrice, // standard price
    uint256[] memory slugs, // item slug with custom price
    uint256[] memory prices
  ) payable ERC721(name, symbol) {
```





Please help me to solve this problem.











Regards,















Don

**Reply by schaable** [reply]
To pass arrays or other complex arguments, you need to have a file that exports the list of arguments, like this:

```
module.exports = [
  "BrandWallet Test001", // name
  "brandwallet-test001", // symbol
  1000, // _royaltyFee
  "0x3029Cf3FC7c62293700f5851d33eF8A867f95dbE", // _royaltyRecipient
  "", // baseTokenUri 
  "", // _collectionSize
  "", // _tokenPrice
  [], // slugs
  [], // prices
];
```

Then, you can run the command with:

npx hardhat verify --network mumbai 0x5462AB3A7c0B434741345b8a64e7B017701C9e57 --constructor-args path-to-your-file.js

**Reply by super0605** [reply]
Hello schaable

Of course, I know this way to use file to figure out the complex arguments.
But I have also another issue to use the file.
If I deploy the smart contract at once, that works for me.
But I am going to deploy the smart contract dynamically.
It means, I need to deploy the many smart contracts with different complex arguments through out our company website(product) in real-time.
In that case, I am not able to create this file based on different complex arguments dynamically in real-time.
So, I’ve used the verify command line instead of using file.

Please help me to solve this with the best solution.


Regards,

Don
